pkgs/development/compilers/chicken/5/overrides.nix

```
@@ -16,7 +16,11 @@ let
     (addPkgConfig old) // (addToPropagatedBuildInputs pkg old);
   broken = old: { meta = old.meta // { broken = true; }; };
   brokenOnDarwin = old: { meta = old.meta // { broken = stdenv.isDarwin; }; };
-in {
+  addToCscOptions = opt: old: {
+    CSC_OPTIONS = lib.concatStringsSep " " ([ old.CSC_OPTIONS or "" ] ++ lib.toList opt);
```

List are already concatenated when serialized ?
```suggestion
    CSC_OPTIONS = [ old.CSC_OPTIONS or "" ] ++ lib.toList opt;
```

---

And make opt a list ?

---

Yes, but I'd rather type the environment(-only) variable as a string, or even rename it to `env.CSC_OPTIONS` (which is not possible now).

---

You can define as env.CSC_OPTION and call `lib.concatStringsSep` only once in eggDerivation.nix

---

Ah it is already done: https://github.com/NixOS/nixpkgs/blob/b9612fb909c94d51eb3090e06ac7cbc4bda2304d/pkgs/development/compilers/chicken/5/eggDerivation.nix#L23

---

Yeah, but I guess `cscOptions` isn't really useful in `overrideAttrs`?